src/Symfony/Bundle/SecurityBundle/Form/Type/UsernamePasswordType.php

```diff
@@ -42,31 +41,23 @@ public function __construct(RequestStack $requestStack)
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder
-            ->add('username', 'Symfony\Component\Form\Extension\Core\Type\TextType')
-            ->add('password', 'Symfony\Component\Form\Extension\Core\Type\PasswordType')
+            ->add('_username', 'Symfony\Component\Form\Extension\Core\Type\TextType')
```

Comment:
It heavily relies on the parameter configuration in security, you probably end up with `createNamedForm('', UsernamePasswordType::class)`, again, depending on your security configuration.

So i think this is only viable, if a) it's fully driven by security (it just works) and b) it's customizable.

Did i mention remember me services already? Imo. this should add a checkbox if needed...